model/user.go

```diff
@@ -47,6 +47,77 @@ const (
 	USER_PASSWORD_MAX_LENGTH  = 72
 )
 
+// This is a struct for a user interaction
+// A user interaction will represent one row in the results of the
+// GetUserInteractions query in store/sqlstore/user_store.go
+// **Note: Using pointers as property types allows for null values
+type UserInteraction struct {
+	// The type of interaction
+	// enum('Reaction', 'Mention', 'Reply', 'DirectMessage')
+	Interaction_Type *string `json:"Interaction_Type"`
+
+	// The username of the user that this interaction took place with
+	Username *string `json:"Username"`
+
+	// Denotes whether or not the current user was the recipient of this interaction
+	Is_Recipient *bool `json:"Is_Recipient"`
+
+	// The context in which this interaction occured (either 'Course' or the learning group type's prefix (ex. 'plg'))
+	Context *string `json:"Context"`
+
+	// The type of channel that this interaction took place in
+	// enum('P', 'O', 'D', 'G')
+	//     P = private channel
+	//     O = public channel
+	//     D = direct message (1 on 1)
+	//     G = group direct message
+	Channel_Type *string `json:"Channel_Type"`
+
+	// The name of the channel that this interaction took place in
+	Channel_Name *string `json:"Channel_Name"`
+}
+
+// This is a struct for a learning group type
+// Each learning group type is derived from personal channel names found in the config
+type LearningGroupType struct {
+	// ex. 'capstone'
+	Prefix string
+}
+
+// This is a struct for all user analytics data sets
+// This is the struct that gets returned to mattermost-redux in the users/{userId}/teams/{teamId}/useranalytics request
+// **Note: Using pointers as property types allows for null values
+type UserAnalytics struct {
+	UserInteractions   []*UserInteraction
+	UserLearningGroups []*LearningGroup
+	LearningGroupTypes []*LearningGroupType
+	// Add more datasets here
+}
```

Comment:
Added TargetUser as a property of UserAnalytics

Reply:
Note that UserAnalytics is the struct that gets returned to redux and it may be redundant, if only for now, to store the TargetUser in redux, since we should know the userId that we sent. However, it might make sense to store it along with the rest of the user analytics data anyway, in case we are fetching for more than one user in the future.

Reply:
Yes, I realized when I wrote this that the caller "knows" what the target user is, but I think adding it to the structure makes it more self-contained which I can see being useful. Thanks for adding it.

Reply:
Yeah makes sense. We can deal with it on the redux end now! I didn't include it in the list of datasets, but it will get returned alongside with the other datasets. It's possible that we would want: 

```
users: {
    analytics: [
         {
          TargetUser: userId1,
          UserInteractions: {},
          etc: {}
          },
          {
          TargetUser: userId2,
          UserInteractions: {},
          etc: {}
          },
     ]
}
```

Right now it is (in the current mattermost-redux PR):
```
users: {
    analytics: {
         TargetUser: userId1,
          UserInteractions: {},
          etc: {}
     }
}
```

Reply:
But we can discuss that on the other PR